internal/cache/cache.go

```diff
@@ -54,20 +64,59 @@
 	if timeout <= 0 {
 		timeout = r.defaultExpiration
 	}
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if r.cache == nil {
+		return
+	}
+
 	r.cache.Set(key, value, time.Duration(timeout)*time.Second)
 }
 
 // Delete cache item by key.
 func (r *repository) Delete(key string) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if r.cache == nil {
+		return
+	}
+
 	r.cache.Delete(key)
 }
 
 // Count returns number of items in the cache.
 func (r *repository) Count() int {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+
+	if r.cache == nil {
+		return 0
+	}
+
 	return r.cache.ItemCount()
 }
 
 // Flush flushes all the items from the cache.
 func (r *repository) Flush() {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if r.cache == nil {
+		return
+	}
+
 	r.cache.Flush()
 }
```

Comment:
**Suggestion:** Consider checking if the cache is nil before calling the Flush method. This can prevent a panic if the cache is already closed.
```suggestion
func (r *repository) Flush() {
	r.mu.Lock()
	defer r.mu.Unlock()

	if r.cache != nil {
		r.cache.Flush()
	}
}
```